src/frontend/src/handler/create_sink.rs

```diff
@@ -12,84 +12,144 @@
+        name: table.name,
+        columns: table.columns,
+        pk: table.pk,
+        dependent_relations: table.dependent_relations,
+        distribution_key: table.distribution_key,
+        stream_key: table.stream_key,
+        appendonly: table.appendonly,
+        properties: table.properties,
+        owner: table.owner,
+        definition: table.definition,
     }
 }
 
+pub fn gen_sink_query_from_name(from_name: ObjectName) -> Result<Query> {
+    let table_factor = TableFactor::Table {
+        name: from_name,
+        alias: None,
+    };
+    let from = vec![TableWithJoins {
+        relation: table_factor,
+        joins: vec![],
+    }];
+    let select = Select {
+        from,
+        projection: vec![SelectItem::Wildcard],
+        ..Default::default()
+    };
+    let body = SetExpr::Select(Box::new(select));
+    Ok(Query {
+        with: None,
+        body,
+        order_by: vec![],
+        limit: None,
+        offset: None,
+        fetch: None,
+    })
+}
+
 pub fn gen_sink_plan(
     session: &SessionImpl,
     context: OptimizerContextRef,
     stmt: CreateSinkStatement,
 ) -> Result<(PlanRef, ProstSink)> {
     let db_name = session.database();
-    let (schema_name, associated_table_name) =
-        Binder::resolve_schema_qualified_name(db_name, stmt.materialized_view.clone())?;
+    let (sink_schema_name, sink_table_name) =
+        Binder::resolve_schema_qualified_name(db_name, stmt.sink_name.clone())?;
+
+    let (query, sink_col_names) = match stmt.sink_from {
+        CreateSink::From(from_name) => {
+            let (from_schema_name, from_table_name) =
+                Binder::resolve_schema_qualified_name(db_name, from_name.clone())?;
+            let (_, _, from_catalog) =
+                session.get_table_catalog_for_create(from_schema_name, &from_table_name)?;
+            let from_col_names = from_catalog
+                .columns()
+                .iter()
+                .filter_map(|catalog| {
+                    (!catalog.is_hidden()).then_some(catalog.column_desc.name.clone())
+                })
+                .collect_vec();
+            (
+                Box::new(gen_sink_query_from_name(from_name)?),
+                Some(from_col_names),
+            )
+        }
+        CreateSink::AsQuery(query) => (query, None),
+    };
+
+    let (sink_database_id, sink_schema_id) =
+        session.get_database_and_schema_id_for_create(sink_schema_name)?;
 
-    let (database_id, schema_id, associated_table_catalog) =
-        session.get_table_catalog_for_create(schema_name, &associated_table_name)?;
+    let definition = query.to_string();
+
+    let bound = {
+        let mut binder = Binder::new(session);
+        binder.bind_query(*query)?
+    };
+
+    // If colume names not specified, use the name in materialized view.
+    let col_names = get_column_names(&bound, session, stmt.columns)?.or(sink_col_names);
 
-    let sink_name = Binder::resolve_sink_name(stmt.sink_name)?;
```

Comment:
Help to remove `resolve_sink_name`. Thanks

Reply:
maybe no need to do that. still not sure if it will be used later

Reply:
This `resolve_sink_name` was used to ensure that the sink and the corresponding table are under the same schema. I think after your PR, we don't need this constraint anymore